### DISCUSSION TITLE
Feature request : Improve fast folder navigation

### DISCUSSION THREAD

**Discussion by jaroet** [mikekazakov/nimble-commander#146] [Ideas]
Thanks for making Nimble Commander. It is fast and very nice to work with. 

I would like to request an additional feature (or expansion of an existing feature) to improve on the folder navigation. My problem is that as I am very familiair with the folder structure in my computer I now where I want to go and I want to do that with the least amount of keystrokes. 

History and favorites are perfect for that (also with the search within that dialog) but it is limitied to like 9 entries. While I would appreciate a lot more. If the history would be unlimited that would be a great way to do what I want. Hit CMD-2 or 3, start typing part of the foldername, select the correct one from the limited list and be on my way. Off course the history should be remembered between program runs and partial results should be included in the resultset. 

The downfall of this is that you need to visit the folder first (at least for 1 time) to get it in the history. 

Another solution is to use the goto folder dialog to include a search for foldernames only. Like hitting CMD-SHFT-G to get to the dialog and start typing. While typing the list with results gets smaller and smaller untill it is small enough to pick the result you want. Hit ENTER and you are in the place you want to be. 

Both options can be based on history (preferably global history) that is stored between sessions or based on a folder search only. Something like ' mdfind kind:folder "searchvalue" ' in the terminal. 

It would even be better if the list with search results is sorted on the number of activations. So the most used folder are shown at the top of the list, respecting the searchvalue of course. 

That would make navigation to folders even more simple then it is now already. 

why I do not have enough with the current solutions:
- history is not stored between sessions
- Favorites and Frequently visited is limited to only 9 values. An alternative would be to be able to search all recently visited locations. 
- I have lots of folder I quickly  want to go to: like client folders, git repo folders, the usual OS folders like (documents, downloads, pictures, ect), music folder with lots of folders with artistname. 

One program that does this very well is https://fman.io. It is even mentioned as the first feature that sets it apart on the homepage. 

Let me know what you think about this proposal. 

Thank you and greetings,
Jeroen

**Comment by Robert-M-Muench** [comment]
I want to 2nd this overall idea of more **frequently visited** entries.

Some ideas:
1. https://atuin.sh/ - How about integrating with this? This would link my CLI navigation with NC navigation, and **search while you type** fits perfectly. Since this tracks the complete CLI filtering CD commands would be necessary, but doable.
2. How about letting me switch folders like https://github.com/ajeetdsouza/zoxide from NC? That would be very useful. A simple way to hack in some path parts and have NC jump to it.

Neither option would enforce any limitations on the number of favorites
